cli/test/test_runner_slurm.py

```diff
@@ -24,14 +28,16 @@ def tearDown(self):
 
     def test_tail_output(self):
         self.Popen.set_command('tail -f slurm-x.out', returncode=0)
-        self.slurm_runner._tail_output('slurm-x.out')
-        # TODO: assert that tail was invoked
+        self.assertEqual(
+            self.slurm_runner._tail_output('slurm-x.out'), 0)
+        self.assertEqual(len(self.slurm_runner._out_stream_pid), 1)
 
     def test_stop_running_tasks(self):
         self.Popen.set_command('scancel --name job_a', returncode=0)
         self.slurm_runner._spawned_jobs.add('job_a')
+        # If no exception is raised, implies scancel was called and
+        # got caught in mock Popen.
```

Comment:
this is not necessarily true. The ideal is to check by invoking the `self.Popen.all_calls()` to ensure `scancel` got called.